sqlserver/datadog_checks/sqlserver/connection.py

```diff
@@ -156,6 +156,7 @@ def open_db_connections(self, db_key, db_name=None, is_default=True, key_prefix=
             else:
                 cs += self._conn_string_odbc(db_key, db_name=db_name)
                 rawconn = pyodbc.connect(cs, timeout=self.timeout, autocommit=True)
+                rawconn.timeout = self.timeout
```

Comment:
does setting this the timeout arg above not actually work?

Reply:
The timeout passed to the "connect" function is the connection timeout, which is different from the request timeout that is being set on the following line.

Reply:
ah ... how .... fun

Reply:
No, that's pyodbc only. I added tests that verify that the request timeout works as expected and it is working in the windows tests, which uses `adodbapi`.